staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -151,34 +180,27 @@ type priorityLevelState struct {
 }
 
 // NewTestableController is extra flexible to facilitate testing
-func newTestableController(
-	informerFactory kubeinformers.SharedInformerFactory,
-	flowcontrolClient flowcontrolclient.FlowcontrolV1beta1Interface,
-	serverConcurrencyLimit int,
-	requestWaitLimit time.Duration,
-	obsPairGenerator metrics.TimedObserverPairGenerator,
-	queueSetFactory fq.QueueSetFactory,
-) *configController {
+func newTestableController(config TestableConfig) *configController {
```

Comment:
I like this change to config, but for me it's completely independent from the changes of controller fights.

@MikeSpreitzer - would you be able to split that part to a separate PR? It would make both the review easier as well as potential revert in case of problems with one of the changes more targeted.

Reply:
[Another argument - I would be happy to just approve this cleanup immediately, whereas for the other changes, we definitely want Daniel or David to take a look too.]

Reply:
Done

Reply:
Thanks - that PR has already merged. Can you please rebase this one?

Reply:
done